From e967aeaf77ff416427385ae954e388233192bcb0 Mon Sep 17 00:00:00 2001
From: Padmini Vijayakumar <padvij@codeaurora.org>
Date: Thu, 30 Oct 2014 06:37:21 +0000
Subject: [PATCH 2/2] Deferring of services from init.qcom.rc and
 init.qcom.usb.rc scripts for faster boot

Change-Id: Id6b2a87a0c932a5a194fe3ea5eb995e1fb0e0951
Signed-off-by: Baji Patthan <bpatth@codeaurora.org>

Conflicts:
	rootdir/etc/init.qcom.rc
	rootdir/etc/init.qcom.usb.rc
---
 common.mk                    |  2 +-
 rootdir/etc/init.qcom.rc     | 57 +++++++++++++++++++++++++++++++++++++-------
 rootdir/etc/init.qcom.usb.rc |  8 ++++++-
 3 files changed, 57 insertions(+), 10 deletions(-)

diff --git a/common.mk b/common.mk
index 1d6513a..1f47ae4 100644
--- a/common.mk
+++ b/common.mk
@@ -475,7 +475,7 @@ PRODUCT_PACKAGES += $(BSON)
 PRODUCT_PACKAGES += $(BT)
 PRODUCT_PACKAGES += $(C2DCC)
 PRODUCT_PACKAGES += $(CONNECTIVITY)
-PRODUCT_PACKAGES += $(CHARGER)
+#PRODUCT_PACKAGES += $(CHARGER)
 PRODUCT_PACKAGES += $(CURL)
 PRODUCT_PACKAGES += $(DASH)
 PRODUCT_PACKAGES += $(E2FSPROGS)
diff --git a/rootdir/etc/init.qcom.rc b/rootdir/etc/init.qcom.rc
index e204eec..2893659 100755
--- a/rootdir/etc/init.qcom.rc
+++ b/rootdir/etc/init.qcom.rc
@@ -61,6 +61,7 @@ service modem-sh /system/bin/sh /system/etc/init.qcom.modem_links.sh
     class core
     user root
     oneshot
+    disabled
 
 on boot
     chown bluetooth bluetooth /sys/module/bluetooth_power/parameters/power
@@ -300,6 +301,7 @@ service qcom-c_main-sh /system/bin/sh /init.class_main.sh
     class main
     user root
     oneshot
+    disabled
 
 on property:vold.decrypt=trigger_restart_framework
     start qcom-c_main-sh
@@ -308,24 +310,27 @@ on property:vold.decrypt=trigger_restart_framework
 service cnd /system/bin/cnd
     class late_start
     socket cnd stream 660 root inet
+    disabled
 
 service irsc_util /system/bin/logwrapper /system/bin/irsc_util "/etc/sec_config"
     class main
     user root
     oneshot
+    disabled
 
 service rmt_storage /system/bin/rmt_storage
     class core
     user root
     disabled
 
-on property:ro.boot.emmc=true
-    start rmt_storage
+#on property:ro.boot.emmc=true
+#    start rmt_storage
 
 service config_bluetooth /system/bin/sh /system/etc/init.qcom.bt.sh "onboot"
     class core
     user root
     oneshot
+    disabled
 
 service hciattach /system/bin/sh /system/etc/init.qcom.bt.sh
     class late_start
@@ -532,6 +537,7 @@ service lowi-server /system/bin/lowi-server
     class late_start
     user gps
     group gps net_admin wifi inet
+    group gps qcom_oncrpc inet net_raw qcom_diag net_admin wifi
     disabled
 
 service fm_dl /system/bin/sh /system/etc/init.qcom.fm.sh
@@ -565,6 +571,7 @@ service qcom-sh /system/bin/sh /init.qcom.sh
     class late_start
     user root
     oneshot
+    disabled
 
 service qcom-post-boot /system/bin/sh /system/etc/init.qcom.post_boot.sh
     class late_start
@@ -586,14 +593,32 @@ service wifi-crda /system/bin/sh /system/etc/init.crda.sh
 
 
 on property:sys.boot_completed=1
-    write /proc/bootkpi/marker_entry "Boot completed"
+#    write /proc/bootkpi/marker_entry "Boot completed"
     start qcom-post-boot
+    start qcom-c_main-sh
+    start modem-sh
+    start rmt_storage
+    start cnd
+    start irsc_util
+    start config_bluetooth
+    start loc_launcher
+    start atfwd
+    start ppd
+    start sdcard
+    start fuse_sdcard
+    start fuse_usbotg
+    start fuse_sata
+    start ssr_diag
+    start qcom-sh
+    wait /sys/class/net/can0
+    start can
 
 service atfwd /system/bin/ATFWD-daemon
     class late_start
     user system
     group system radio
     onrestart /system/bin/log -t RIL-ATFWD -p w "ATFWD daemon restarted"
+    disabled
 
 service hdmid /system/bin/hdmid
     class late_start
@@ -605,6 +630,7 @@ service ppd /system/bin/mm-pp-daemon
     user system
     socket pps stream 0660 system system graphics
     group system graphics
+    disabled
 
 service hostapd /system/bin/hostapd -dddd /data/hostapd/hostapd.conf
     class late_start
@@ -661,9 +687,23 @@ service fuse_sdcard /system/bin/sdcard -u 1023 -g 1023 -d /mnt/media_rw/sdcard1
     class late_start
     disabled
 
-on property:ro.fuse_sdcard=true
-    start sdcard
-    start fuse_sdcard
+#on property:ro.fuse_sdcard=true
+#    start sdcard
+#    start fuse_sdcard
+
+service fuse_usbotg /system/bin/sdcard -u 1023 -g 1023 -w 1023 -d /mnt/media_rw/usbotg /storage/usbotg
+    class late_start
+    disabled
+
+service fuse_sata /system/bin/sdcard -u 1023 -g 1023 -w 1023 -d /mnt/media_rw/sata /storage/sata
+    class late_start
+    disabled
+
+service can /system/bin/sh /init.qti.can.sh
+   class late_start
+   user root
+   disabled
+   oneshot
 
 # Binding fuse mount point to /storage/emulated/legacy
 on property:init.svc.sdcard=running
@@ -683,13 +723,14 @@ service hcidump /system/bin/sh /system/etc/hcidump.sh
     disabled
     oneshot
 
-service charger /charger
-    class charger
+#service charger /charger
+#    class charger
 
 service ssr_diag /system/bin/ssr_diag
     class late_start
     user system
     group system
+    disabled
 
 #Disable boot timeout for external modems.
 on property:persist.sys.mdm.timeout_disable=1
diff --git a/rootdir/etc/init.qcom.usb.rc b/rootdir/etc/init.qcom.usb.rc
index c89131c..b16bafa 100644
--- a/rootdir/etc/init.qcom.usb.rc
+++ b/rootdir/etc/init.qcom.usb.rc
@@ -32,6 +32,7 @@ service qcom-usb-sh /system/bin/sh /init.qcom.usb.sh
     class core
     user root
     oneshot
+    disabled
 
 # Following are the parameters required for usb functionality. They provide configurable options like
 # product_id/vendor id and allows specifying required functions:
@@ -337,7 +338,7 @@ on property:sys.usb.config=diag,diag_mdm,serial_hsic,serial_tty,rmnet_hsic,mass_
     write /sys/class/android_usb/android0/f_rmnet/transport_names rmnet_hsic
     write /sys/class/android_usb/android0/functions diag,adb,serial,rmnet,mass_storage
     write /sys/class/android_usb/android0/enable 1
-    start adbd
+#    start adbd
     setprop sys.usb.state ${sys.usb.config}
 
 # Fusion 3 composition with diag_mdm
@@ -655,3 +656,8 @@ on property:sys.usb.config=diag,qdss,adb
     write /sys/class/android_usb/android0/enable 1
     start adbd
     setprop sys.usb.state ${sys.usb.config}
+
+# Configure usb otg mode as per the property value
+on property:sys.boot_completed=1
+    start qcom-usb-sh
+    start adbd
-- 
1.8.2.1

